File: php/features.php
<?php
declare(strict_types=1);

// FakeStore SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FakeStoreFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FakeStoreBaseFeature();
            case "test":
                return new FakeStoreTestFeature();
            default:
                return new FakeStoreBaseFeature();
        }
    }
}

File: py/features.py
# FakeStore SDK feature factory

from feature.base_feature import FakeStoreBaseFeature
from feature.test_feature import FakeStoreTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FakeStoreBaseFeature(),
        "test": lambda: FakeStoreTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
